api/integrations/perforce/zulip_change-commit.py

```diff
@@ -72,13 +72,45 @@
 metadata = git_p4.p4_describe(changelist)  # type: Dict[str, str]
 
 destination = config.commit_notice_destination(changeroot, changelist)  # type: Optional[Dict[str, str]]
+
 if destination is None:
     # Don't forward the notice anywhere
     sys.exit(0)
 
-message = "**{0}** committed revision @{1} to `{2}`.\n\n> {3}".format(
+ignore_missing_stream = None
+try:
+    ignore_missing_stream = config.ZULIP_IGNORE_MISSING_STREAM
+except AttributeError:
+    # old config
+    pass
+
+if ignore_missing_stream:
+    # Check if the destination stream exists yet
+    stream_state = client.get_stream_id(destination["stream"])
+    if stream_state["result"] == "error":
+        # Silently discard the message
+        sys.exit(0)
+
+change = metadata["change"]
+p4web = None
+try:
+    p4web = config.P4_WEB
+except AttributeError:
+    # old config
+    pass
+
+if p4web is not None:
+    # linkify the change number
+    change = '[{1}]({0}/{1}?ac=10)'.format(p4web, change)
+
+message = """**{0}** committed revision @{1} to `{2}`.
+
+```quote
+{3}
+```
+""".format(
     metadata["user"],
-    metadata["change"],
+    change,
     changeroot,
     metadata["desc"])  # type: str
 
```

Comment:
There's still 4 arguments (`metadata["change"]` being replaced with `change` in this patch) but otherwise keeping the existing `{0}` - `{3}` placeholders and 4 arguments.

Hopefully it will be clearer with named placeholders